About and Activities UI refinement

comzeradd opened the pull request:
We no longer have Clubs roles so no need to hard code them in the template. I changed this block to be a `InputTextContentBlock` so now it's dynamic. The background image on the body block is also dynamic now.

This also containers some css adjustments to align with [Phase 1 design](https://github.com/mozilla/coss/issues/322).

comzeradd commented:
I added a commit with UI adjustments to the Activities to meet the Phase 1 Design.

akatsoulas commented on coss/home/static/js/clubs_about.js:
```diff
@@ -0,0 +1,29 @@
+$(document).ready(function() {
```
Why is this moved in the `home` app?

comzeradd replied:
All the javascript code there is needed for the parts of the template that live in the home app. I'll rename it to a more generic name.

comzeradd commented:
PR updated

akatsoulas commented on coss/home/models.py:
```diff
@@ -22,10 +22,18 @@ class AboutSnippet(models.Model):
+    )
     info_title = models.CharField(blank=True, default='', max_length=40)
     info_text = fields.RichTextField(blank=True)
+    details_title = models.CharField(blank=True, default='', max_length=40)
```
There is another `details` section in the streamfield? Are the two related? If yes, this shouldn't part of the snippet page. If not, it's a bit confusing to have details in two fields.

comzeradd replied:
Yes, good point. Since the rest of the details block is also generic and not specific to OS Clubs, I moved it inside that Snippet. Now it's less confusing for a Content creator.

![screenshot-2017-9-6 wagtail - editing about snippet - test](https://user-images.githubusercontent.com/939357/30113226-a6c97342-931c-11e7-8c73-1390de81f94c.png)

akatsoulas commented on coss/home/templates/home/about_page.html:
```diff
@@ -15,46 +15,80 @@
+      {% if page.about.body_image %}
+        {% image page.about.body_image original as bg_photo %}
+      {% endif %}
+      <div id="body" class="body" data-bg-img="{{ bg_photo.url }}">
```
What happens if the `bg_photo` is undefined?

comzeradd replied:
In that case that data attribute is empty and the javascript code will fallback to the default one.

akatsoulas commented on coss/home/templates/home/about_page.html:
```diff
@@ -15,46 +15,80 @@
+      <div id="body" class="body" data-bg-img="{{ bg_photo.url }}">
+        <div class="body-image"></div>
         <div class="well">
+          {% if page.about.body_title %}
```
Shouldn't this `if` be before the `<div class="well"`

comzeradd replied:
Not necessarily. We won't to display the well and `page.about.body_text` even when no title is set.

akatsoulas replied:
Agreed, but the comment stands for the body_text too. It's a not required field. Ideally, as we make the project more generic, we want to only render blocks that have content. In the future, this can be a standalone snippet inserted in a template.
Probably, it's OK for now, but please add at least comments as placeholders on what we need to revisit in order to make the template more generic.

comzeradd replied:
Yes, adding comments is a good idea. Just to point out that this whole block is wrapped inside an if statement (`{% if page.about.body_text %}`), so this is rendered only if a body text is provided.

akatsoulas commented on coss/opensource_clubs/templates/opensource_clubs/about_page.html:
```diff
@@ -10,45 +10,8 @@
-        </div>
-      </div>
-    </div>
+  <div class="collaborators">
```
Since we have the snippet/streamfield for activities, don't we want to add the missing section in this PR?

comzeradd commented:
PR updated. I added the collaborators section. I also modified the model to group relevant fields. This will make it easier for Content creators to understand the scope of each field.

comzeradd commented:
> Found another file with the destination path 'css/_colors.scss'. It will be ignored since only the first encountered file is collected. If this is not what you want, make sure every static file has a unique path.

That doesn't cause any problem. Sass ignores partial files anyway (the ones the begin with _), so it's ok to have the same name on different apps.

akatsoulas commented:
Instead of ignoring it, can't we just fix it?

akatsoulas commented on coss/home/models.py:
```diff
@@ -20,36 +22,80 @@ class Meta:
-    bottom_cta2_text = models.CharField('Text', blank=True, default='', max_length=20)
-    bottom_cta2_link = models.URLField('Link', blank=True, default='')
+    body_title = models.CharField(verbose_name='Title', blank=True, default='',
+                                  max_length=40)
```
Let's be consistent, in other places the title holds `100 chars`.

akatsoulas replied:
The above applies to all the title fields that exist in this PR.

comzeradd replied:
All `title` fields have the same limit. All `text` fields are either unlimited or are limited based on design (eg. buttons can only have really short text).

akatsoulas commented on coss/home/models.py:
```diff
@@ -20,36 +22,80 @@ class Meta:
+                                           max_length=40)
+    collaborators_text = fields.RichTextField(verbose_name='Text', blank=True)
+    collaborators_cta_text = models.CharField(verbose_name='CTA Text', blank=True, default='',
+                                              max_length=20)
```
`20` feels too restrictive here. maybe doubling that number?

comzeradd commented:
Yes I'll make some tests. It seems to be a [known issue](https://github.com/django-compressor/django-compressor/issues/720). Although this is not related to this PR. This warning was introduced by 229efaea1fe27ca534405d1ac2991b26c5c9765d.

akatsoulas commented on coss/home/models.py:
```diff
@@ -20,36 +22,80 @@ class Meta:
+                                  max_length=40)
+    body_text = fields.RichTextField(verbose_name='Text', blank=True)
+    body_image = models.ForeignKey(
+        'wagtailimages.Image',
```
Locally, the image that I chose is not displayed in the template. Please make sure that this is working properly before merging this PR. Also that the fallback is working too.

akatsoulas commented on coss/home/models.py:
```diff
@@ -20,36 +22,80 @@ class Meta:
+    info_text = fields.RichTextField(verbose_name='Text', blank=True)
+    details_title = models.CharField(verbose_name='Title', blank=True, default='',
+                                     max_length=40)
+    details = fields.StreamField([
```
We need to show in the snippet that details is a required field In the admin the user can only see the streamfield item and there is no indication that it's a required field.

comzeradd replied:
Details is not required. We only render this block if we have at least on details block.
